src/coreComponents/constitutive/ConstitutiveManager.cpp

```diff
@@ -67,6 +67,19 @@ ConstitutiveManager::hangConstitutiveRelation( string const & constitutiveRelati
   material->allocateConstitutiveData( *parent,
                                       numConstitutivePointsPerParentIndex );
 
+
+  std::vector<string> const subRelationNames = material->getSubRelationNames();
+
+  for( string const & subRelationName : subRelationNames )
+  {
+    ConstitutiveBase const & subRelation = getConstitutiveRelation( subRelationName );
+
+    std::unique_ptr< ConstitutiveBase > constitutiveModel = subRelation.deliverClone( subRelationName, parent );
```

Comment:
`constitutiveModel` goes out of scope and destructs at the end of the loop body. It needs to be moved a wrapper on the new `constitutiveGroup`, similar to what happens to `material` some lines below.

Reply:
should work fine now. Should I add a check that if the wrapper already exists we don't register it?